relay-server/src/services/processor/ourlog.rs

```diff
@@ -0,0 +1,86 @@
+//! Log processing code.
+use std::sync::Arc;
+
+use crate::envelope::Item;
+use relay_config::Config;
+use relay_dynamic_config::Feature;
+use relay_event_schema::protocol::OurLog;
+
+use crate::envelope::ItemType;
+use crate::services::processor::should_filter;
+use crate::services::projects::project::ProjectInfo;
+use crate::utils::{ItemAction, TypedEnvelope};
+
+use relay_ourlogs::OtelLog;
+use relay_protocol::Annotated;
+
+#[cfg(feature = "processing")]
+use {
+    crate::envelope::ContentType,
+    crate::services::outcome::{DiscardReason, Outcome},
+    crate::services::processor::LogGroup,
+};
+
+/// Removes logs from the envelope if the feature is not enabled.
+pub fn filter<Group>(
+    managed_envelope: &mut TypedEnvelope<Group>,
```

Comment:
nit: We could define a trait `LogLike` implemented and add it as a trait
```suggestion
pub fn filter(
    managed_envelope: &mut TypedEnvelope<LogGroup>,
```